android/src/main/java/com/mattermost/networkclient/APIClientModule.kt

```diff
@@ -1,186 +1,146 @@
 package com.mattermost.networkclient
 
-import android.net.Uri
+import com.mattermost.networkclient.enums.APIClientEvents
+import com.mattermost.networkclient.enums.RetryTypes
 import com.facebook.react.bridge.*
 import okhttp3.*
-import okio.*
-import com.mattermost.networkclient.enums.APIClientEvents
-import com.mattermost.networkclient.helpers.*
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import android.net.Uri
+import android.util.Log
+import com.mattermost.networkclient.helpers.ProgressListener
+import com.mattermost.networkclient.helpers.UploadFileRequestBody
+import java.io.IOException
+import java.lang.Exception
 import kotlin.collections.HashMap
 
 class APIClientModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
-
-    var sessionsClient = mutableMapOf<String, OkHttpClient.Builder>()
-    var sessionsRequest = mutableMapOf<String, Request.Builder>()
-    var calls = mutableMapOf<String, Call>()
+    private val clients = mutableMapOf<HttpUrl, NetworkClient>()
```

Comment:
Something I didn't think of yesterday; I used the singleton pattern because the global interceptor needs an already initialised object where it can call the existing clients / interceptors / configs - I'm wondering if this will still work because NetworkClient is now a class that needs to be initialised in order to be used? Unless there's a way of passing around the NetworkClient class that already exists to the parent app...